### PR TITLE
[improvement] Improved global code

### PR DESCRIPTION
**Pull request by Olyno**
This pull request includes a big rework of the project, with a better code, and a better data split.

**Related to:**

 - snowpackjs/snowpack#1277

### CODE REVIEW

File: src/data/librairies.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        "name": "svelte"
+    },
+    {
+        "name": "vue"
+    },
+    {
+        "name": "blank"
+    }
+]
```
Comment (gr2m) [inline src/data/librairies.json]: typo in the file name? I wonder if `src/data/librairies.json` should have been `src/data/libraries.json`? But then I see you used the term at several places throughout the code base, I guess you prefer French? 😄

**Review by gr2m — COMMENTED** [review]
I tested it but got an error

```
➜  node /Users/gregor/Projects/create-snowpack-app-interactive/bin/index.js 
✔ At what path should we generate the template? … .
✔ What UI library/framework would you like to use? › Preact
✔ Should we add some plugins? › 
✔ What port the app should use? … 8080
✔ Which package manager should we use? › Yarn
⠹ Downloading the selected template...
Something bad happened when generating project :(
⠇ Downloading the selected template...
```

I wonder if you could add more details to the "Something bad happened when generating project" error?

The problems seemed to have been the `.` answer to the `At what path should we generate the template?` question. It worked when I answered `test` instead. Maybe you could add support for `.`?

I also haven't seen the "Would you like to use TypeScript?" question?

File: src/data/librairies.json (hunk 1 above)
Comment (Olyno) [reply src/data/librairies.json]: ahahah sorry for that, I'm french so I shuffled english and french, I fix that xDD

Comment (gr2m) [reply src/data/librairies.json]: pas de problème :)

**Comment by Olyno** [comment]
> I wonder if you could add more details to the "Something bad happened when generating project" error?

Hmm it should show you all the error, it's weird if it doesn't

> Maybe you could add support for .?

Was supported, but got a an error which has been fixed

> I also haven't seen the "Would you like to use TypeScript?" question?

Yes indeed you will maybe not see it in some case. It's a dynamic question. If you select a template which doesn't contain any typescript equivalent, then the question is skipped

**Comment by gr2m** [comment]
> Yes indeed you will maybe not see it in some case. It's a dynamic question. If you select a template which doesn't contain any typescript equivalent, then the question is skipped

Smart 👍🏼 

This looks good, let's ship it and share with the snowpack community for more feedback

**Comment by gr2m** [comment]
🎉 Could you please share on https://github.com/snowpackjs/snowpack/issues/1277 once you release a new version?

**Pull request merged**